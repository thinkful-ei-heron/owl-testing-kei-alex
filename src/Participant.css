* {
  box-sizing: border-box;
}

.Participant {
  display: flex;
  margin-top: 5px;
}

.user-avatar {
  margin-top: 5px;
  width: 30px;
  height: 30px;
  border-radius: 100%;
  font-size: 10px;
}

.dot {

}

.in-session {

}

.hidden {
  
}