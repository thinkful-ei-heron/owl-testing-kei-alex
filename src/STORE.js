const STORE = {
  participants: [
    {
      id: 1,
      name: 'Koren Templeton',
      avatar:
          'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true,
      user: false
    },
    {
      id: 2,
      name: 'Caty Flucker',
      avatar:
          'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true,
      user: true
    },
    {
      id: 3,
      name: 'Axe Kubicka',
      avatar: 'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1',
      inSession: false,
      onStage: false,
      user: false
    },
    {
      id: 4,
      name: 'Frank Runciman',
      avatar:
          'https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: false,
      user: false
    },
    {
      id: 5,
      name: 'Ashla Attwool',
      avatar:
          'https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true,
      user: false
    }
  ]
};

export default STORE;