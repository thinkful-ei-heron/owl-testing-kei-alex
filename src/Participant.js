import React from 'react';
import './Participant.css'

export default function Participant(props) {
  return (
    <div className = "Participant">
      <img className= "user-avatar" src= {props.avatar} alt=""></img>
      <b>{props.name}</b>
      <br></br>
      {props.inSession ? <span className="dot in-session"></span> :  <span className="dot hidden"></span>}
      <span className="status">{props.onStage ? 'on stage' : 'in session'}</span>
    </div>
  )
}