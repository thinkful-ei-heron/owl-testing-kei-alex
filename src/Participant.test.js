import React from 'react';
import ReactDOM from 'react-dom';
import STORE from './STORE';
import Participant from './Participant';
import renderer from 'react-test-renderer';

describe('Participant tests', () => {
  //smoke test
  it('render with prop', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Participant
      id = {STORE.participants[0].id}
      name = {STORE.participants[0].name}
      avatar = {STORE.participants[0].avatar}
      inSession = {STORE.participants[0].inSession}
      onStage = {STORE.participants[0].onStage}
    />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Participant
        id = {STORE.participants[0].id}
        name = {STORE.participants[0].name}
        avatar = {STORE.participants[0].avatar}
        inSession = {STORE.participants[0].inSession}
        onStage = {STORE.participants[0].onStage}
      />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  })
});