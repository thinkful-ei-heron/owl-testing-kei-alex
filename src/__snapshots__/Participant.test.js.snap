// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant tests renders the UI as expected 1`] = `
<div
  className="Participant"
>
  <img
    alt=""
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  <b>
    Koren Templeton
  </b>
  <span
    className="dot in-session"
  />
  <span
    className="status"
  >
    on stage
  </span>
  }
</div>
`;
